### PR TITLE
Update the settings screen's layout

### PR DESCRIPTION
**Pull request by ThomasMPont**
## Overview

I've improved the layout of the settings screen. It should now be more intuitive, and more pleasant to look at.

## Screenshots

Here's the old settings screen for reference,
image

Here's the new look of the settings screen
image

## Changes

Figma was updated to reflect these changes.

The settings screen layout and colors were changed according to [this task](https://github.com/orgs/Signify-epfl/projects/2/views/1?filterQuery=settings&pane=issue&itemId=90615386&issue=Signify-epfl%7Csignify-app%7C419)'s specifications.

The logout button is now in the settings screen's top bar.

The pen icon next to the username field now selects the field on click.

The color of the "cancel" button of confirmation popups was changed on every screen for consistency.

### CODE REVIEW

**Comment by GRGool** [comment]
Hey, checking your branch it seems like its not up to date with main? The N/A language option I removed earlier is in your branch. Not sure how you should fix it. Ill still review the rest.

File: app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt [hunk 1]
```diff
@@ -202,36 +209,43 @@ fun EditableProfilePictureField(userViewModel: UserViewModel, profilePictureUrl:
...
+        modifier = Modifier.background(MaterialTheme.colorScheme.background),
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.Center) {
-          Icon(
-              modifier = Modifier.testTag("editUsernameButton"),
-              imageVector = Icons.Outlined.Edit,
+          val focusRequester = remember { FocusRequester() }
+
+          BasicButton(
+              onClick = { focusRequester.requestFocus() },
```
Comment (GRGool) [inline app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt:244]: Nice, I dont think it was doing anything previously.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt [hunk 2]
```diff
@@ -191,8 +191,8 @@ fun ChallengeModeAlertDialog(
             onClick = onDismiss,
             testTag = "CancelButton",
             text = stringResource(R.string.cancel_text),
-            backgroundColor = MaterialTheme.colorScheme.surface,
-            textColor = MaterialTheme.colorScheme.onSurface,
+            backgroundColor = MaterialTheme.colorScheme.onSurface,
+            textColor = MaterialTheme.colorScheme.surface,
```
[the remark this answers is not in the document]
Comment (GRGool) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt:195]: Also feels weird to have to review a change in CreateAChallenge and FriendList when the PR is about the setting's screen layout.

**Review by GRGool — COMMENTED** [review]
Hey. Everything looks good, the changes are appreciated. I had positive opinions on most of the changes, I have put some small comments below. We are trying to get as many things fixed as possible but try to remain in scope of the PR, theres changes to the common BasicButton, so multiple screens, as well as specifics on the createChallenge and FriendList which is not super confortable to review. When the issue with the N/A option still being in is resolved I'll approve

**Review by GRGool — CHANGES_REQUESTED** [review]
Once the out of date branch issue is fixed ill approve as im unsure if merging this will just reintroduce issues

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThomasMPont) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt:195]: I wanted the other screens to have the right color for the cancel button too. I probably should have opened a separate PR.

**Comment by ThomasMPont** [comment]
> Once the out of date branch issue is fixed ill approve as im unsure if merging this will just reintroduce issues

The reason features are missing is because I haven't pulled the latest changes from main. Github would block my PR if there was merge conflicts. But there isn't any. So the branch can automatically be merged without any issues.

If you still want to check, as a reviewer, just switch to my branch and merge with main. You can even push to it.
I've done it for you this time.

I don't thing we should be avoiding automatic merges, as they only rarely break stuff on `main`. I believe this has unnecessarily delayed this PR.

**Review by GRGool — APPROVED** [review]

**Comment by GRGool** [comment]
> > Once the out of date branch issue is fixed ill approve as im unsure if merging this will just reintroduce issues
> 
> The reason features are missing is because I haven't pulled the latest changes from main. Github would block my PR if there was merge conflicts. But there isn't any. So the branch can automatically be merged without any issues.
> 
> If you still want to check, as a reviewer, just switch to my branch and merge with main. You can even push to it. I've done it for you this time.
> 
> I don't thing we should be avoiding automatic merges, as they only rarely break stuff on `main`. I believe this has unnecessarily delayed this PR.

I agree, just wanted to be sure. Also you still have the CI running on a commit for 5h +

**Pull request merged**